src/fusor/translator.py

```diff
@@ -0,0 +1,586 @@
+        self, chrom1: str, chrom2: str, descr: str | None = None
+    ) -> CausativeEvent:
+        """Infer Causative Event. Currently restricted to rearrangements
+
+        :param chrom1: The chromosome for the 5' partner
+        :param chrom2: The chromosome for the 3' partner
+        :param descr: An annotation describing the fusion event. This input is supplied to the eventDescription CausativeEvent attribute.
+        :return: A CausativeEvent object if construction is successful
+        """
+        if descr and "rearrangement" in descr:
+            return CausativeEvent(
+                eventType=EventType("rearrangement"), eventDescription=descr
+            )
+        if chrom1 != chrom2:
+            return CausativeEvent(eventType=EventType("rearrangement"))
+        return None
+
+    def _get_gene_element_unnormalized(self, symbol: str) -> GeneElement:
+        """Return GeneElement when gene symbol cannot be normalized
+
+        :param symbol: A gene symbol for a fusion partner
+        :return: A GeneElement object
+        """
+        return (
+            GeneElement(
+                type="GeneElement",
+                gene={
+                    "id": f"gene:{symbol}",
+                    "label": symbol,
+                    "type": "Gene",
+                },
+            ),
+            None,
+        )
+
+    def _get_gene_element(self, genelist: str, caller: Caller) -> GeneElement:
+        """Return a GeneElement given an individual/list of gene symbols and a
+        fusion detection algorithm
+
+        :param genelist: A gene symbol or list of gene symbols
+        :param caller: The examined fusion detection algorithm
+        :return A GeneElement object
+        """
+        if "," not in genelist or caller != caller.ARRIBA:
+            ge = self.fusor.gene_element(gene=genelist)
+            return ge if ge[0] else self._get_gene_element_unnormalized(genelist)
+
+        genes = genelist.split(",")
+        dists = []
+        for gene in genes:
+            start, end = gene.rfind("("), gene.rfind(")")
+            dists.append(int(gene[start + 1 : end]))
+        gene = (
+            genes[0].split("(")[0] if dists[0] <= dists[1] else genes[1].split("(")[0]
+        )
+        ge = self.fusor.gene_element(gene=gene)
+        return ge if ge[0] else self._get_gene_element_unnormalized(gene)
+
+    def _get_genomic_ac(self, chrom: str, build: Assembly) -> str:
+        """Return a RefSeq genomic accession given a chromosome and a reference build
+        :param chrom: A chromosome number
+
+        :param build: The assembly, either GRCh37 or GRCh38
+        :return The corresponding refseq genomic accession
+        """
+        sr = self.fusor.cool_seq_tool.seqrepo_access
+        if build == Assembly.GRCH37:
+            alias_list, errors = sr.translate_identifier(f"GRCh37:{chrom}")
+        else:
+            alias_list, errors = sr.translate_identifier(f"GRCh38:{chrom}")
+        if errors:
```

Comment:
Is it confirmed that build will be GRCh37 or GRCh38? If so, we can just do: `alias_list, errors = sr.translate_identifier(f"{build}:{chrom}")` for both. Otherwise:

```suggestion
            alias_list, errors = sr.translate_identifier(f"{build}:{chrom}")
        else:
            alias_list, errors = sr.translate_identifier(f"{Assembly.GRCH38.value}:{chrom}")
        if errors:
```